libs/langchain-core/src/_standard/message.ts

```diff
@@ -0,0 +1,1119 @@
+    return content
+      .filter((block) => block.type === "text")
+      .map((block) => block.text)
+      .join("");
+  }
+}
+
+/** Parameters for creating a SystemMessage */
+type SystemMessageParams<
+  TStructure extends $MessageStructure = $StandardMessageStructure
+> = {
+  /** Optional unique identifier for the message */
+  id?: string;
+  /** Optional name/identifier for the system that generated this message */
+  name?: string;
+  /** Optional metadata associated with the system message, as defined by the message structure */
+  metadata?: $InferMessageProperty<TStructure, "system", "metadata">;
+} & (
+  | {
+      /** Array of content blocks that make up the message */
+      content: Array<$InferMessageContent<TStructure, "system">>;
+    }
+  | {
+      /** Text content for the message */
+      content: string;
+    }
+);
+
+/**
+ * Represents a system message that provides context, instructions, or configuration to the AI.
+ *
+ * System messages are typically used to set the behavior, personality, or operational parameters
+ * for an AI assistant. They are usually processed before user messages and help establish
+ * the context for the conversation.
+ *
+ * @template TStructure - The message structure type that defines the content and property types.
+ *                        Defaults to $StandardMessageStructure.
+ *
+ * @example
+ * ```ts
+ * // Create a system message with simple text
+ * const systemMessage = new SystemMessage("You are a helpful assistant.");
+ *
+ * // Create a system message with structured content
+ * const systemMessageWithContent = new SystemMessage([
+ *   { type: "text", text: "You are a helpful assistant. " },
+ *   { type: "text", text: "Always be polite and concise." }
+ * ]);
+ *
+ * // Access the combined text content
+ * console.log(systemMessage.text); // "You are a helpful assistant."
+ * ```
+ */
+export class SystemMessage<
+  TStructure extends $MessageStructure = $StandardMessageStructure
+> implements Message<TStructure, "system">, $BrandedMessage
+{
+  /** @internal */
+  [__BRANDED_MESSAGE]: never;
+
+  /** The message type, always "system" for system messages */
+  readonly type = "system" as const;
+
+  /** Unique identifier for this message */
+  id: string;
+
+  /** Optional name/identifier for the system that generated this message */
+  name?: string;
+
+  /** Array of content blocks that make up the message content */
+  content: Array<$InferMessageContent<TStructure, "system">>;
+
+  /** Metadata associated with the system message */
+  metadata?: $InferMessageProperty<TStructure, "system", "metadata">;
+
+  /**
+   * Creates a system message with simple text content.
+   * @param text - The text content for the message
+   */
+  constructor(text: string);
+  /**
+   * Creates a system message with structured content blocks.
+   * @param content - Array of content blocks for the message
+   */
+  constructor(content: Array<$InferMessageContent<TStructure, "system">>);
```

Comment:
_nit_: Do we want to have this overload? B/c of that we're now suggesting every method from `Array` in IntelliSense